lib/active_merchant/billing/gateways/beanstream.rb

```diff
@@ -215,6 +217,22 @@ def scrub(transcript)
       def build_response(*args)
         Response.new(*args)
       end
+
+      def add_three_ds(post, options)
+        return unless three_d_secure = options[:three_d_secure]
+
+        post[:SecureXID] = (three_d_secure[:ds_transaction_id] || three_d_secure[:xid]) if three_d_secure.slice(:tres, :cuatro).values.any?
```

Comment:
🤔 Do these keywords actually work? They don't seem to in irb. In any case, I don't think we should rely on the order of elements in this hash for logic.

Reply:
Ohhh my bad on that @curiousepic , it should be 

`three_d_secure.slice(:ds_transaction_id, :xid).values.any?`

going to fix that inmediatly

basically the idea is to use transaction _id is present what I asume is 90%+ of the cases and if no present the fallback to xid.